bundles/org.openhab.binding.fenecon/src/main/java/org/openhab/binding/fenecon/internal/api/FeneconController.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.fenecon.internal.api;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.fenecon.internal.FeneconConfiguration;
+import org.openhab.binding.fenecon.internal.exception.FeneconAuthenticationException;
+import org.openhab.binding.fenecon.internal.exception.FeneconCommunicationException;
+import org.openhab.binding.fenecon.internal.exception.FeneconException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link FeneconController} class provides API access to the FENECON system.
+ *
+ * @author Philipp Schneider - Initial contribution
+ */
+@NonNullByDefault
+public class FeneconController {
+
+    private final Logger logger = LoggerFactory.getLogger(FeneconController.class);
+
+    private final FeneconConfiguration config;
+    private final HttpClient httpClient;
+    private final Builder baseHttpRequest;
+
+    public FeneconController(FeneconConfiguration config) {
+        this.config = config;
+
+        logger.debug("FENECON: initialize REST-API connection to {} with polling interval: {} sec", getBaseUrl(config),
+                config.refreshInterval);
+
+        httpClient = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(10)).build();
```

Comment:
A specific reason for not using the common httpclient ?

Reply:
I always think it's good to integrate as few external libs as possible, especially when Java provides solutions for this.  In this case, the Java client is perfectly suitable and automatically receives security fixes via the Java update.

Reply:
Perfectly sane 'rule'. For openHAB we like to control threads, resources and have made some design choices and that include the usage of the common httpclient unless there is some binding specific condition that can not be met with the common client.
So it would be nice if you inject the client and re-use that.

Reply:
I have replaced the Java client with the Jetty http client.